File: Java/String1/Left2.java
/*
 * CodingBat Java Activity "left2" from String-1 section.
 * codingbat.com
 */

public class Left2 
{
	private static int failures = 0;
	private static final String METHOD_NAME = "left2";
	
	/*
	 * INSTRUCTIONS
	 * 
	 * Given a string, return a "rotated left 2" version where the first 2 
	 * chars are moved to the end. The string length will be at least 2.
	 * 
	 * left2("Hello") -> "lloHe"
	 * left2("java") -> "vaja"
	 * left2("Hi") -> "Hi"
	 */
	public static String left2(String str)
	{
		//put your code ONLY here!
	}
	
	public static void main(String[] args)
	{
		testCase("\"Hello\"", "\"lloHe\"");
		testCase("\"java\"", "\"vaja\"");
		testCase("\"Hi\"", "\"Hi\"");
		testCase("\"code\"", "\"deco\"");
		testCase("\"cat\"", "\"tca\"");
		testCase("\"12345\"", "\"34512\"");
		testCase("\"Chocolate\"", "\"ocolateCh\"");
		testCase("\"bricks\"", "\"icksbr\"");
		
		System.out.println();
		System.out.println(failures + " test cases failed.");
	}

	private static void testCase(String a, String result)
	{
		String r;
		try
		{
			r = ""+ "\"" + left2(a.substring(1,a.length()-1)) + "\"";
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.println("FAIL: ");
				failures++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.print("FAIL: ");
			failures++;
		}
		System.out.println(METHOD_NAME+"("+a+") -> "+r + ", EXPECTED: "+result);
	}

}

File: Python/string-1/left2.py
'''
CodingBat Python Activity "left2" from string-1.
codingbat.com
'''

failures = 0

def left2(str):
    '''
Given a string, return a "rotated left 2" version where the first 2 chars are moved to the end.
The string length will be at least 2.


left2('Hello') → 'lloHe'
left2('java') → 'vaja'
left2('Hi') → 'Hi'
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("left2('Hello')", 'lloHe')
    catchFailures("left2('java')", 'vaja')
    catchFailures("left2('Hi')", 'Hi')
    catchFailures("left2('code')", 'deco')
    catchFailures("left2('cat')", 'tca')
    catchFailures("left2('12345')", '34512')
    catchFailures("left2('Chocolate')", 'ocolateCh')
    catchFailures("left2('bricks')", 'icksbr')

        
    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()
